packages/shared/src/contexts/payment/Paddle.tsx

```diff
@@ -1,212 +1,24 @@
 import type { ReactElement } from 'react';
-import React, { useCallback, useMemo } from 'react';
-import type { PaddleEventData } from '@paddle/paddle-js';
-import { CheckoutEventNames } from '@paddle/paddle-js';
-import { useQuery } from '@tanstack/react-query';
-import { useRouter } from 'next/router';
-import { useAuthContext } from '../AuthContext';
-import { plusSuccessUrl } from '../../lib/constants';
-import { usePlusSubscription } from '../../hooks';
-import { feature } from '../../lib/featureManagement';
-import { PlusPriceType, PlusPriceTypeAppsId } from '../../lib/featureValues';
-import { getPrice } from '../../lib';
-import { useFeature } from '../../components/GrowthBookProvider';
-import type {
-  OpenCheckoutProps,
-  PaymentContextData,
-  PaymentContextProviderProps,
-  ProductOption,
-} from './context';
-import { PaymentContext } from './context';
-import { usePaddle } from '../../features/payment/hooks/usePaddle';
+import React from 'react';
+import type { CheckoutEventNames, PaddleEventData } from '@paddle/paddle-js';
+import { usePaddlePayment } from '../../hooks/usePaddlePayment';
+import type { PaymentContextProviderProps } from './context';
+import { BasePaymentProvider } from './BasePaymentProvider';
+
+export type PaddleSubProviderProps = PaymentContextProviderProps<
+  PaddleEventData,
+  CheckoutEventNames
+>;
 
 export const PaddleSubProvider = ({
   children,
   successCallback,
 }: PaymentContextProviderProps): ReactElement => {
-  const router = useRouter();
```

Comment:
Almost everything is managed within the `usePaddlePayment`, then everything related to product pricing is handled within the `BasePaymentContext` which uses `useProductPricing` that fetch the already organized data.